FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Azure/Blob/AzureBlobContainerClient.cs

```diff
@@ -249,5 +266,62 @@ public async Task<bool> ReleaseLeaseAsync(string blobName, string leaseId, Cance
                 return false;
             }
         }
+
+        public async Task MoveDirectory(string sourceDirectory, string targetDirectory, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var sourceDirectoryClient = _dataLakeFileSystemClient.GetDirectoryClient(sourceDirectory);
+
+                // Create target parent directory if not exists.
+                var targetParentDirectory = Path.GetDirectoryName(targetDirectory);
+                if (!string.IsNullOrEmpty(targetParentDirectory))
+                {
+                    var targetParentDirectoryClient = _dataLakeFileSystemClient.GetDirectoryClient(targetParentDirectory);
+                    await targetParentDirectoryClient.CreateIfNotExistsAsync(cancellationToken: cancellationToken);
+                }
+
+                await sourceDirectoryClient.RenameAsync(targetDirectory);
```

Comment:
What will happen if targetDirectory already exists?

Reply:
The rename operation will fail and AzureBlobOperationFailedException will be thrown. Since it's an atomic operation, we don't expect to move a folder twice here.